src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,29 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    while (true) {
+      comments();
+
+      switch (sym) {
+        case EndLiteralArray:
+          expect(EndLiteralArray, DelimiterClosingTag.class);
+          return createLiteralArray(expressions, getSource(coord));
+        case Period:
```

Comment:
hmm, I think this is not entirely correct.

This is more like: `{ DOT* expr*  }`.

This means it parses `{....3....}` and it is `{3}`, which is surprising, I think.

And, it also parses `{3 3 3 3}`, which is `{3. 3. 3. 3}`. This should also not be supported.